pkg/node/types/node.go

```diff
@@ -81,6 +87,49 @@ func ParseCiliumNode(n *ciliumv2.CiliumNode) (node Node) {
 		}
 	}
 
+	cidrs := make([]*cidr.CIDR, 0, len(n.Status.ENI.Subnets))
+	for _, subnet := range n.Status.ENI.Subnets {
+		if cidr, err := cidr.ParseCIDR(subnet); err == nil {
+			cidrs = append(cidrs, cidr)
+		}
+	}
+	node.IPv4NativeRoutingCIDRs = cidrs
+
+	interfaces := make([]Interface, 0, len(n.Status.ENI.ENIs))
+	for _, eni := range n.Status.ENI.ENIs {
+		gateway, err := subnetGatewayAddress(eni.Subnet)
+		if err != nil {
+			continue
+		}
+
+		mac, err := mac.ParseMAC(eni.MAC)
+		if err != nil {
+			continue
+		}
+
+		endpointAddresses := make([]Address, 0, len(eni.Addresses))
+		for _, address := range eni.Addresses {
+			endpointAddress := Address{
+				Type: addressing.NodeInternalIP, // FIXME is this correct?
```

Comment:
We could probably just use `net.IP` instead of `Address` as we'll never use the `Type`.